### PR TITLE
Fix flaky test

### PR DESCRIPTION
**Pull request by penge**
Fixes flaky **'Sniff interval'** in `sniff.test.js`.

The problem was caused by not waiting for the asynchronous `kill()` to finish the task,
before making a request about the current `connectionPool.size`.

The error was:

```js
t.strictEqual(client.connectionPool.size, 3)

// expected: 3  => because one node should be killed
//   actual: 4  => because the request was made before the node was killed
```

The solution was to update `kill()` by adding a `callback` and calling it after the `stop()` in `stoppable.js` has finished. This was then used in `sniff.test.js` and `resurrect.test.js`.

Logging in `shutdown()` is also improved.

Unused `spawn()` is removed.

Closes https://github.com/elastic/elasticsearch-js/issues/1108.

### CODE REVIEW

**Comment by penge** [comment]
Here it is @delvedor ;)

**Comment by penge** [comment]
I've added some more `setTimeout` calls to test that the **SNIFF** event is only run when:

```
1) delay is greater than sniffInterval
2) delay is greater than time of last sniff + sniffInterval
```

This is also mentioned in the added comments.

I think it's now good to merge.

File: test/behavior/sniff.test.js [hunk 1]
```diff
@@ -111,39 +111,44 @@ test('Should handle hostnames in publish_address', t => {
...
-      client.info(t.error)
+      kill('node1', () => {
+        client.info(t.error) // meets 1) and 2)
+      })
     }, 150)
 
-    setTimeout(() => {
-      t.strictEqual(client.connectionPool.size, 3)
-    }, 200)
-
```
Comment (delvedor) [inline test/behavior/sniff.test.js:146]: Great catch! I would kindly ask you to revert this change, leave the test as it was before and keep just the `kill` fix :)

Comment (penge) [reply test/behavior/sniff.test.js:146]: Thanks! Can you please explain to me why to revert the test?

I see 4 reasons why the new test is better. The previous test would _NOT_ work (1), would not be complete (2), can randomly fail (3), and has an unrelated assertion (4).

Here's the detail:

1) By not waiting for `kill()` to finish, `shutdown()` would try to kill 4 nodes instead of 3, because of outdated `nodes`. This would result in the following error:

```
kill('node1')
> kill node1

shutdown()
> kill node0
> kill node1  // Should NOT be here! Cannot kill!
> kill node2
> kill node3

Error [ERR_SERVER_NOT_RUNNING]: Server is not running.
```

2) The previous test does not test **SNIFF** event earlier than `sniffInterval` and earlier than "time of last sniff" + `sniffInterval`.

3) The previous statement:

```js
setTimeout(() => {
  t.strictEqual(client.connectionPool.size, 3)
}, 200)
```

Has an expectation that depends on **SNIFF** to run `this.connectionPool.update(hosts)`. Also depends on lazy loading of `info`, connection speed, time of killing the node.

4) Similar to point 3). This test is primarily about **'Sniff interval'** and not about nodes. Also, `client.on(events.SNIFF, (err, result) => { ... }` already has the assertion needed, and compares 4 nodes with 4, on the first run, and 3 with 3, on the second run.

Please explain. I don't see the reasons to revert the test.

**Comment by penge** [comment]
No problem. Cool
